packages/@glimmer/core/src/managers/modifier.ts

```diff
@@ -25,12 +39,152 @@ export function capabilities(
-  updateModifier(instance: ModifierInstance, args: Args): void;
-  destroyModifier(instance: ModifierInstance, args: Args): void;
+  createModifier(definition: unknown, args: Args): ModifierStateBucket;
+  installModifier(instance: ModifierStateBucket, element: Element, args: Args): void;
+  updateModifier(instance: ModifierStateBucket, args: Args): void;
+  destroyModifier(instance: ModifierStateBucket, args: Args): void;
 }
 
 export type ModifierDefinition<_Instance = unknown> = {};
+
+export type SimpleModifier = (element: Element, ...args: unknown[]) => undefined | (() => void);
+
+interface SimpleModifierStateBucket {
+  definition: SimpleModifier;
+  destructor?(): void;
+  element?: Element;
+}
+
+class SimpleModifierManager implements ModifierManager<SimpleModifierStateBucket> {
+  capabilities = capabilities('3.13');
+
+  createModifier(definition: SimpleModifier, args: Args): SimpleModifierStateBucket {
+    if (DEBUG) {
+      assert(Object.keys(args.named).length === 0, `You used named arguments with the ${definition.name} modifier, but it is a standard function. Normal functions cannot receive named arguments when used as modifiers.`);
+    }
+
+    return { definition };
+  }
+
+  installModifier(bucket: SimpleModifierStateBucket, element: Element, args: Args): void {
+    bucket.destructor = bucket.definition(element, ...args.positional);
+    bucket.element = element;
+  }
+
+  updateModifier(bucket: SimpleModifierStateBucket, args: Args): void {
+    this.destroyModifier(bucket);
+    this.installModifier(bucket, bucket.element!, args);
+  }
+
+  destroyModifier(bucket: SimpleModifierStateBucket): void {
+    const { destructor } = bucket;
+
+    if (destructor !== undefined) {
+      destructor();
+    }
+  }
+}
+
+const SIMPLE_MODIFIER_MANAGER = new SimpleModifierManager();
+
+///////////
+
+export class CustomModifierState<ModifierStateBucket> {
+  public tag = createUpdatableTag();
+
+  constructor(
+    public element: SimpleElement,
+    public delegate: ModifierManager<ModifierStateBucket>,
+    public modifier: ModifierStateBucket,
+    public argsProxy: Args,
+    public capturedArgs: CapturedArguments
+  ) {}
+
+  destroy(): void {
+    const { delegate, modifier, argsProxy } = this;
+    delegate.destroyModifier(modifier, argsProxy);
+  }
+}
+
+export class CustomModifierManager<ModifierStateBucket>
+  implements
+    VMModifierManager<
+      CustomModifierState<ModifierStateBucket>,
+      ModifierDefinition<ModifierStateBucket>
+    > {
+  create(
+    element: SimpleElement,
+    definition: ModifierDefinition<ModifierStateBucket>,
+    args: VMArguments,
+    dynamicScope: DynamicScope
+  ): CustomModifierState<ModifierStateBucket> {
+    const owner = dynamicScope.get(OWNER_KEY).value() as object;
+    let delegate = getModifierManager(owner, definition);
+
+    if (delegate === undefined) {
+      if (DEBUG) {
+        assert(
```

Comment:
Curious, do these get stripped properly on their own or do they have to be within an `if (DEBUG) {`? Also, if they do have to be in an `if (DEBUG) {` what happens to the import?

Reply:
I believe the import should be tree-shaken, which is why I figured this pattern would be alright for now. We currently only replace `DEBUG`.